src/components/SelectBox/SelectBox.tsx

```diff
@@ -0,0 +1,302 @@
+import _ from 'lodash'
+
+const StyledDropdown = styled.div<{ width?: string }>`
+  position: relative;
+  flex: 1;
+  width: ${(props) => props.width || '100%'};
+
+  display: flex;
+  flex-direction: column;
+`
+
+const StyledDropdownToggle = styled.div`
+  cursor: pointer;
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  align-items: center;
+  border-radius: 999px;
+  background-color: ${themeFunc('elevation', '9')};
+  padding: 9px 14px;
+  gap: 8px;
+  border: 1px solid;
+  ::after {
+    display: none;
+  }
+  :focus {
+    box-shadow: none !important;
+    outline: none;
+  }
+`
+
+const StyledIconChevronBottom = styled(Chevron_bottom)`
+  transition: transform 0.2s ease-in-out;
+  position: absolute;
+  right: 16px;
+  width: 16px;
+  height: 16px;
+  fill: ${themeFunc('gray', '0')};
+`
+const StyledIconindent = styled(Indent_DownRight)`
+  transition: transform 0.2s ease-in-out;
+  margin-right: 12px;
+  width: 16px;
+  height: 16px;
+  fill: ${themeFunc('gray', '0')};
+`
+const dropdownKeyframes = keyframes`
+  0% {
+    opacity: 0;
+    transform: scaleY(0);
+  }
+  80% {
+    opacity: 1;
+    transform: scaleY(1.1);
+  }
+  100% {
+    transform: scaleY(1);
+  }
+`
+const StyledDropdownMenu = styled.div<{ maxHeight?: string }>`
+  box-sizing: border-box;
+  position: absolute;
+  top: 40px;
+  border-radius: 14px;
+  padding: 8px 0;
+  cursor: pointer;
+  justify-content: center;
+  animation: ${dropdownKeyframes} 0.3s ease-out;
+  transform-origin: top;
+  width: 100%;
+  z-index: 1;
+  background-color: ${themeFunc('gray', '10')};
+  box-shadow: 0px 4px 12px 0px rgba(25, 26, 28, 0.05);
+  max-height: ${(props) => props.maxHeight || 'none'};
+  overflow-y: ${(props) => (props.maxHeight ? 'auto' : 'visible')};
+`
+const View = styled.div`
+  display: flex;
+  flex-direction: column;
+`
+const Row = styled(View)`
+  flex-direction: row;
+  align-items: center;
+`
+
+const StyledDropdownItem = styled(View)<{
+  isSelected?: boolean
+  level: number
+}>`
+  justify-content: center;
+  position: relative;
+  padding: ${(props) =>
+    props.isSelected
+      ? '6px 0px'
+      : props.level === 0
+        ? '14px 16px'
+        : `14px ${16 * props.level}px`};
+  color: ${themeFunc('gray', '10')};
+  font-weight: 400;
+  :hover {
+    background-color: ${themeFunc('elevation', '8')};
+  }
+  :active {
+    font-weight: 700;
+  }
+  :first-of-type {
+    border-top: 0;
+  }
+`
+const NeonBar = styled.div`
+  width: 4px;
+  height: 36px;
+  margin-right: 12px;
+  background-color: ${themeFunc('brand', '5')};
+  transition: background-color 0.3s ease;
+  border-radius: 6px;
+  justify-content: center;
+`
+interface Item {
+  id: string
+  label: string
+  subItems?: Item[]
+  isDisabled?: boolean
+  img?: string
+}
+
+export interface KaSelectBoxProps {
+  optionList: Item[]
+  onSelect: (value: string) => void
+  selectedValue: string
+  placeholder?: string
+  indentIcon?: boolean
+  width?: string
```

Comment:
provide `containerStyle` instead of `width` so that users can custom the style by themselves

https://github.com/kaiachain/kaia-design-system/blob/main/src/components/Input/index.tsx#L94